src/crowdin_gui.cpp

```diff
@@ -538,6 +516,35 @@ class CrowdinOpenDialog : public wxDialog
 } // anonymous namespace
 
 
+dispatch::future<void> CrowdinUpdateCatalogIDsFromHeaders(CatalogPtr catalog)
+{    
+    if(catalog->GetCrowdinProjectId() > 0 && catalog->GetCrowdinFileId() > 0)
+        return dispatch::make_ready_future();
+
+    const auto projName = catalog->Header().GetHeader("X-Crowdin-Project");
+    auto filePathName = catalog->Header().GetHeader("X-Crowdin-File");
+
+    if(!filePathName.StartsWith(L"/"))
+        filePathName.insert(0, L'/');
+
+    return CrowdinClient::Get().GetUserProjects().then([=](std::vector<CrowdinClient::ProjectListing> projects)
+    {
+        auto projIt = find_if(projects.begin(), projects.end(),
+                                [=](const auto& proj) { return projName == proj.identifier; });
+        if(projIt == projects.end())
+            throw Exception(_("File project not found for currently authorized user"));
+        return CrowdinClient::Get().GetProjectInfo(projIt->id).then([=](CrowdinClient::ProjectInfo proj)
+        {
+            auto fileIt = find_if(proj.files.begin(), proj.files.end(),
+                                [=](const auto& file) { return filePathName == file.pathName; });
+            if(fileIt == proj.files.end())
+                throw Exception(_("File path not found in it project in currently authorized user"));
```

Comment:
The two messages have inconsistent phrasing ("in user", "for user"), lack punctuation and aren't very clear (what is "file project"?).

It would be better to throw a single, same error text, which explains to the user the error is Crowdin related and is because the user is not authorized to access the file. For example:

"You are not authorized to sync this file to Crowdin."

Details of why (project or file ID mismatch, error obtained etc.) are an appropriate use of `wxLogTrace`.

Reply:
Addressed in 3792a79dc0ceae1329e70a2790f14e3c4ad3653f by making above error messages really clear, informative, precise and fully explanatory what IMO still might be really useful for customers to understand what happened, what the problem is and how to solve it (without looking into logs) while they does not load minds of customers too much by any technical details or so. Though please feel free to `Unresolve` this conversation if you are still disagree.